habitat_baselines/common/utils.py

```diff
@@ -174,3 +226,92 @@ def generate_video(
         tb_writer.add_video_from_np_images(
             f"episode{episode_id}", checkpoint_idx, images, fps=fps
         )
+
+
+def image_resize_shortest_edge(img, size: int, channels_last: bool = False):
+    """Resizes an img so that the shortest side is length of size while
+        preserving aspect ratio.
+
+    Args:
+        img: the array object that needs to be resized (HWC) or (NHWC)
+        size: the size that you want the shortest edge to be resize to
+        channels: a boolean that channel is the last dimension
+    Returns:
+        The resized array as a torch tensor.
+    """
+    no_batch_dim = len(img.shape) == 3
+    if len(img.shape) < 3 or len(img.shape) > 5:
+        raise NotImplementedError()
+    img = _to_tensor(img)
+    if no_batch_dim:
+        img = img.unsqueeze(0)  # Adds a batch dimension
+    if channels_last:
+        # NHWC
+        h, w = img.shape[-3:-1]
+        if len(img.shape) == 4:
+            img = img.permute(0, 3, 1, 2)
+        else:
+            img = img.permute(0, 1, 4, 2, 3)
+    else:
+        # NCHW
+        h, w = img.shape[-2:]
+
+    # Percentage resize
+    if w > h:
```

Comment:
maybe
`percent = size / max(h, w)`

Reply:
it's shortest resize so it would percent = size / min(h,w)

Reply:
Yes, you're right.